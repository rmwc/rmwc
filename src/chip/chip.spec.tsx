import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Chip, ChipSet } from './';

describe('Chip', () => {
  it('renders', () => {
    const { asFragment } = render(
      <ChipSet>
        <Chip icon="favorite" trailingIcon="close" label="test-label" />
      </ChipSet>
    );

    expect(screen.getByText('test-label')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with children', () => {
    render(
      <ChipSet>
        <Chip icon="favorite" trailingIcon="close">
          test-label
        </Chip>
      </ChipSet>
    );

    expect(screen.getByText('test-label')).toBeInTheDocument();
  });

  it('handles selected', () => {
    render(
      <ChipSet>
        <Chip checkmark selected label="test-label" />
        <Chip checkmark selected icon="favorite" label="test-label" />
      </ChipSet>
    );
    expect(screen.getAllByRole('row')[0]).toHaveClass('mdc-chip--selected');
    expect(screen.getAllByRole('row')[1]).toHaveClass('mdc-chip--selected');
  });

  it('handles onInteraction', async () => {
    const onInteraction = jest.fn();
    render(<Chip label="my label" onInteraction={onInteraction} />);

    userEvent.click(screen.getByText('my label'));

    await waitFor(() => expect(onInteraction).toHaveBeenCalledTimes(1));
  });

  it('handles custom ChipIcon', () => {
    render(<Chip icon="favorite" />);

    expect(screen.getByText('favorite')).toBeInTheDocument();
  });

  it('handles onTrailingIconInteraction', async () => {
    let onInteraction = jest.fn();
    let onRemove = jest.fn();

    render(
      <Chip
        trailingIcon="close"
        onTrailingIconInteraction={onInteraction}
        onRemove={onRemove}
      />
    );

    userEvent.click(screen.getByText('close'));

    await waitFor(() => {
      expect(onInteraction).toHaveBeenCalledTimes(1);
    });
  });
});
