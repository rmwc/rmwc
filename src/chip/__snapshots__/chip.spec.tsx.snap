// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders 1`] = `
<DocumentFragment>
  <div
    class="mdc-chip-set"
  >
    <div
      class="mdc-ripple-surface mdc-chip"
      role="row"
    >
      <div
        class="mdc-chip__ripple"
      />
      <i
        class="rmwc-icon rmwc-icon--ligature material-icons mdc-chip__icon mdc-chip__icon--leading"
      >
        favorite
      </i>
      <span
        role="gridcell"
      >
        <span
          class="mdc-chip__text__primary-action"
          role="button"
          tabindex="0"
        >
          <span
            class="mdc-chip__text"
          >
            test-label
          </span>
        </span>
      </span>
      <i
        class="rmwc-icon rmwc-icon--ligature material-icons mdc-chip__icon mdc-chip__icon--trailing"
        role="button"
        tabindex="0"
      >
        close
      </i>
    </div>
  </div>
</DocumentFragment>
`;
